core/services/ocr2/plugins/ocr2keeper/custom_telemetry.go

```diff
@@ -0,0 +1,127 @@
+	"github.com/smartcontractkit/chainlink/v2/core/services/synchronization/telem"
+	"github.com/smartcontractkit/chainlink/v2/core/static"
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+)
+
+const (
+	customTelemChanSize = 100
+)
+
+type AutomationCustomTelemetryService struct {
+	utils.StartStopOnce
+	monitoringEndpoint commontypes.MonitoringEndpoint
+	headBroadcaster    httypes.HeadBroadcaster
+	headCh             chan blockKey
+	unsubscribe        func()
+	chDone             chan struct{}
+	lggr               logger.Logger
+}
+
+// NewAutomationCustomTelemetryService creates a telemetry service for new blocks and node version
+func NewAutomationCustomTelemetryService(me commontypes.MonitoringEndpoint, hb httypes.HeadBroadcaster, lggr logger.Logger) *AutomationCustomTelemetryService {
+	return &AutomationCustomTelemetryService{
+		monitoringEndpoint: me,
+		headBroadcaster:    hb,
+		headCh:             make(chan blockKey, customTelemChanSize),
+		chDone:             make(chan struct{}),
+		lggr:               lggr.Named("Automation Custom Telem"),
+	}
+}
+
+// Start starts Custom Telemetry Service, sends 1 NodeVersion message to endpoint at start and sends new BlockNumber messages
+func (e *AutomationCustomTelemetryService) Start(context.Context) error {
+	return e.StartOnce("AutomationCustomTelemetryService", func() error {
+		e.lggr.Infof("Starting: Custom Telemetry Service")
+		vMsg := &telem.NodeVersion{
+			Timestamp:   uint64(time.Now().UTC().UnixMilli()),
+			NodeVersion: static.Version,
+		}
+		wrappedVMsg := &telem.AutomationTelemWrapper{
+			Msg: &telem.AutomationTelemWrapper_NodeVersion{
+				NodeVersion: vMsg,
+			},
+		}
+		bytes, err := proto.Marshal(wrappedVMsg)
+		if err != nil {
+			e.lggr.Errorf("Error occurred while marshalling the Node Version Message %s: %v", wrappedVMsg.String(), err)
+		} else {
+			e.monitoringEndpoint.SendLog(bytes)
+			e.lggr.Infof("NodeVersion Message Sent to Endpoint: %d", vMsg.Timestamp)
+		}
+		_, e.unsubscribe = e.headBroadcaster.Subscribe(&headWrapper{e.headCh})
+		go func() {
+			e.lggr.Infof("Started: Custom Telemetry Service")
+			for {
+				select {
+				case blockInfo := <-e.headCh:
+					blockNumMsg := &telem.BlockNumber{
+						Timestamp:   uint64(time.Now().UTC().UnixMilli()),
+						BlockNumber: uint64(blockInfo.block),
+						BlockHash:   blockInfo.hash,
+					}
+					wrappedBlockNumMsg := &telem.AutomationTelemWrapper{
+						Msg: &telem.AutomationTelemWrapper_BlockNumber{
+							BlockNumber: blockNumMsg,
+						},
+					}
+					b, err := proto.Marshal(wrappedBlockNumMsg)
+					if err != nil {
+						e.lggr.Errorf("Error occurred while marshalling the Block Num Message %s: %v", wrappedBlockNumMsg.String(), err)
+					} else {
+						e.monitoringEndpoint.SendLog(b)
+						e.lggr.Infof("BlockNumber Message Sent to Endpoint: %d", blockNumMsg.Timestamp)
+					}
+				case <-e.chDone:
+					return
+				}
+			}
+		}()
+		return nil
+	})
```

Comment:
start a new go routine to query latest config details from registry at 1 min interval